core/src/main/java/org/apache/iceberg/avro/Avro.java

```diff
@@ -206,10 +207,22 @@ public ReadBuilder rename(String fullName, String newName) {
       return this;
     }
 
+    /**
+     * Field name to field Id map, constructed from the table properties
+     * The current mapping is from name to field id. Aliases of a column
+     * are stored as separate entries in the map with the same value
+     * TODO: Address case insenstivity
+     * TODO: could a schema have duplicate column names?
```

Comment:
No. We assume that for any name like a.b.c there is only one field. It could be ['a.b', 'c'], ['a', 'b', 'c'], ['a', 'b.c'] or ['a.b.c'], but no two of them are allowed.